### ISSUE TITLE
Call for Contributing Data Science Tutorials

### ISSUE THREAD

**Issue by omarsar** [dair-ai/dair-ai.github.io#14]
If you are interested in contributing to democratizing AI research, education, and technologies by publishing tutorials, please propose your ideas below. We will work closely before publishing any data science related tutorials so as to improve standardization, approachability, accessibility, quality, accuracy, and transparency of your work.

To simplify things, we can first work together to publish in this [Medium publication](https://medium.com/dair-ai), and then everything will be reposted to [dair.ai](https://dair.ai/) to improve accessibility. If you prefer to directly post on dair.ai, you can submit a PR with your tutorial, following the format used in these [posts](https://github.com/dair-ai/dair-ai.github.io/tree/master/_posts).

Here is a [tutorial example](https://medium.com/dair-ai/building-rnns-is-fun-with-pytorch-and-google-colab-3903ea9a3a79) that can give you some ideas of the components that are necessary to publish a great and successful tutorial. And here is a [guide](https://colab.research.google.com/drive/1Bs_t7mdo1qoHVxq-Ny7rYiJkYaGEKZtW) that I wrote for helping out data scientists to communicate about their projects.

If you have any ideas on what you want to work on, write them below and we can further discuss more about the process. Ideally, I would prefer tutorials around a widely used and current tool or release/feature, but you can put your suggestions below. Once we have decided to move forward with the project, we can open a separate issue and track the progress via this master issue.

Below I will also post some ideas of tutorials that I have in mind. If you are interested in any of those, you can also work on those. Note that notebooks are the preferred format for tutorials accompanied by a nice summary of it in the form of a blog post.

Feel free to reach out at ellfae@gmail.com or [DM on Twitter](https://twitter.com/omarsar0) for more information on how to contribute. 🙏 

**Ideas:**
- [ ] Fundamentals to NLP (ongoing project, see #30)
- [ ] Introductory deep neural network tutorials with [JAX](https://github.com/google/jax)
- [ ] Introductory NLP tutorials with PyTorch/TensorFlow/spaCy/etc.
- [ ] Introductory Computer Vision tutorials with open source libraries
- [ ] Tutorials based on [Transformers](https://github.com/huggingface/transformers) for different NLP tasks
- [ ] Tutorials on features releases for libraries like Transformers, TensorFlow, PyTorch, etc.
- [ ] Primers for different ML and NLP libraries are very useful to the community

Here are some examples of tutorials that have been successful in our publication:
- [Fundamentals of NLP (Chapter 1): Tokenization, Lemmatization, Stemming, and Sentence Segmentation](https://medium.com/dair-ai/fundamentals-of-nlp-chapter-1-tokenization-lemmatization-stemming-and-sentence-segmentation-b362c5d07684)
- [Primer for Learning Google Colab](https://medium.com/dair-ai/primer-for-learning-google-colab-bb4cabca5dd6)
- [Building RNNs is Fun with PyTorch and Google Colab](https://medium.com/dair-ai/building-rnns-is-fun-with-pytorch-and-google-colab-3903ea9a3a79) 
- [Medical Imaging Analysis using PyTorch](https://medium.com/dair-ai/medical-imaging-analysis-mri-cnn-pytorch-4877e64e7303)
- [A Simple Neural Network from Scratch with PyTorch and Google Colab](https://medium.com/dair-ai/a-simple-neural-network-from-scratch-with-pytorch-and-google-colab-c7f3830618e0)
- [A Gentle Introduction to PyTorch 1.2](https://medium.com/dair-ai/pytorch-1-2-introduction-guide-f6fa9bb7597c)
- [Neural Machine Translation with Attention Using PyTorch](https://medium.com/dair-ai/neural-machine-translation-with-attention-using-pytorch-a66523f1669f)
- [Deep Learning Based Emotion Recognition with PyTorch and TensorFlow](https://medium.com/dair-ai/deep-learning-based-emotion-recognition-with-pytorch-and-tensorflow-61e831f72234)

**Comment by abdulrahimq** [comment]
I like this [roadmap](https://github.com/graykode/nlp-roadmap) as a base to work off. I saw that a tutorial on tokenization, lemmatization, and stemming was done but I think I can add more details to that too and also add examples to what you can do for language like Arabic as it isn’t a trivial problem. I can write about different tokenization techniques for a morphological like Arabic. I can show each different stemming algorithm in more detail for example. I would be able to write around 300 words a day or something along these lines.

I’m thinking of doing basic procedures -> NLP basic hypothesis -> string distance -> graph -> document and in the end word embedding and sequential labeling. Do you think this is a good plan?

**Comment by omarsar** [comment]
Hi @abdulrahimq! These are all great ideas and of course, it would be nice to have content around it, especially for the Arabic language. There is always a lot to learn there. I propose that we open an issue for this task and we can have a deeper discussion about it on Slack or on the issue. In the end, when we have all this material, we can propose a roadmap for learning that consists of notebooks and other beginner materials. That's the ultimate goal and you are spot on. Open the issue here and assign it to yourself and me. 

https://github.com/dair-ai/nlp_fundamentals/issues

**Comment by omarsar** [comment]
Move to Wiki
